admin/pre_reg/views.py

```diff
@@ -20,45 +21,63 @@
 
 get_draft_or_error = functools.partial(get_or_http_error, DraftRegistration)
 
+def get_prereg_drafts(user=None):
+    prereg_schema = MetaSchema.find_one(
+        Q('name', 'eq', 'Prereg Challenge') &
+        Q('schema_version', 'eq', 2)
+    )
+    query = (
+        Q('registration_schema', 'eq', prereg_schema) &
+        Q('approval', 'ne', None)
+    )
+    if user:
+        pass
+        # TODO: filter by assignee; this requires multiple levels of Prereg admins-
+        # one level that can see all drafts, and another than can see only the ones they're assigned.
+        # As a followup to this, we need to make sure this applies to approval/rejection/commenting endpoints
+        # query = query & Q('_metaschema_flags.assignee', 'eq', user._id)
+    return DraftRegistration.find(query)
 
 def is_in_prereg_group(user):
     """Determines whether a user is in the prereg_group
     :param user: User wanting access to prereg material
     :return: True if prereg False if not
     """
-    return user.groups.filter(name='prereg_group').exists()
-
+    return user.is_in_group('prereg_group')
 
 @login_required
-# @user_passes_test(is_in_prereg_group)
-def prereg(request):
+@user_passes_test(is_in_prereg_group)
+def prereg(request, page_number):
     """Redirects to prereg page if user has prereg access
     :param request: Current logged in user
     :return: Redirect to prereg page with username, reviewers, and user obj
     """
-    #prereg_admin = request.user.has_perm('auth.prereg_admin')
-    #user = {
-    #     'username': str(request.user.username),
-    #     'admin': json.dumps(prereg_admin)
-    # }
-    user = {
-        'username': 'user_placeholder',
-        'admin': 'admin_placeholder'
-    }
-    # reviewers = MyUser.objects.prereg_users()
-    reviewers = ['admin_placeholder']
+    paginator = Paginator(get_prereg_drafts(), 5)
+
+    try:
+        page_number = int(page_number or request.GET.get('page'))
+    except (TypeError, ValueError):
+        page_number = 1
+
+    page = paginator.page(page_number)
+
+    try:
+        drafts = [serializers.serialize_draft_registration(d, json_safe=False) for d in page]
+    except EmptyPage:
+        drafts = []
+
+    for draft in drafts:
```

Comment:
An alternative to this would be django's forests.

You should be able to do:
```python
from django.forms import formset_factory
from somewhere import DraftRegistrationForm

DraftRegistrationFormset = formset_factory(DraftRegistrationForm, initial=drafts, extra=0) # extra=0 tells it not to put a blank on in
context = {
    formset=DraftRegistrationFormset,
    ...
    ...
}
```

Then you should be able to loop through formset like a list in the template, each iteration will give you a form.

https://docs.djangoproject.com/en/1.9/topics/forms/formsets/#using-initial-data-with-a-formset